### ISSUE TITLE
[Bug] ...

### ISSUE THREAD

**Issue by jeblad** [stuarthayhurst/privacy-menu-extension#34]
**Describe the issue**
Clicking on the buttons for mic and cam does not shut them off.

**To Reproduce**
Steps to reproduce the behaviour:
1. Attach Camera
2. Check camera in Cheese (or some other app) that the live feed is there
3. Go to Quick Settings
4. Click on Camera
5. Observe the feed in Cheese (or some other app) that the live feed is still there (it should have been blocked)

**Expected behaviour**
Video should be shut off.

**Screenshots and logs**

This is tested with a pretty old Microsoft Studio LifeCam Studio and a mic that reports as a C-Media Electronics Blue Snowball.

How the quick setting menu appeared while using the extension.

![Skjermbilde fra 2023-07-31 23-39-26](https://github.com/stuarthayhurst/privacy-menu-extension/assets/4649804/0de89d8d-fa35-4812-aa8d-33f68063e6f2)

Attached USB devices.

```
john@chimera:~$ lsusb
Bus 008 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 007 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 006 Device 002: ID 174c:3074 ASMedia Technology Inc. ASM1074 SuperSpeed hub
Bus 006 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 005 Device 003: ID 046d:c548 Logitech, Inc. USB Receiver
Bus 005 Device 002: ID 174c:2074 ASMedia Technology Inc. ASM1074 High-Speed hub
Bus 005 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 004 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 003 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 004: ID 05e3:0608 Genesys Logic, Inc. Hub
Bus 001 Device 003: ID 0b05:19af ASUSTek Computer, Inc. AURA LED Controller
Bus 001 Device 002: ID 0b05:1a52 ASUSTek Computer, Inc. USB Audio
Bus 001 Device 008: ID 045e:0772 Microsoft Corp. LifeCam Studio
Bus 001 Device 005: ID 0e8d:0608 MediaTek Inc. Wireless_Device
Bus 001 Device 009: ID 0d8c:0005 C-Media Electronics, Inc. Blue Snowball
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
```

**System information (please complete the following information):**
 - Distro: Ubuntu 23.04
 - GNOME version: 44.3
 - Extension version: 17

**Comment by stuarthayhurst** [comment]
Is this a duplicate of #13?

If the application isn't sandboxed (snap / flatpak), the settings won't do anything.

If they're flatpaked and the settings don't do anything, as long as the settings in GNOME Settings match the extension, any bug isn't extension side.

**Comment by jeblad** [comment]
Ouch, … I was about to check whether you actually disabled the camera and mic.

If all the extension do is to  manipulate rights for some subset of apps, then it isn't a truly privacy protecting extension, it is only a quick access to some rights for those apps. When those settings aren't enforced then the extension gives a false impression of privacy.

You should change the extension so it clearly gives the user the mental idea that this extension manipulates rights for sandboxed apps, and that any apps at system level is still able to access the camera and/or mic.

I would propose that you add a toggle list of which individual apps it is able to limit, with an additional entry for "all apps", "all sandboxed apps", or something similar. The additional toggles will then give the impression that this is at application level and not at the system level. Then I would add "rights" in the name to clearly signal that this isn't a system manipulating extension, it only manipulates the rights for some applications.

**Comment by stuarthayhurst** [comment]
On GNOME Extensions, the extension's name is "Privacy Quick Settings". This is because the extension provides quick access to GNOME's privacy settings.

The URL on GitHub is leftover from when the extension created its own dropdown menu, from the top panel, now put behind a setting.

I intend to keep the privacy name, as it only exposes GNOME's privacy settings in a quicker way, so I feel that using another name is misleading. If GNOME changes their naming, I'd change the extension to match.

The paragraph at the top of this project's README was aimed to help clarify this, I think I used the issue you linked to check it was factually accurate. If I've made a mistake, feel free to point it out :)
